pkg/file/file_test.go

```diff
@@ -90,3 +91,46 @@ func TestExists(t *testing.T) {
 	require.NoError(t, err)
 	assert.False(t, exists)
 }
+
+// TestExtractInstallerMetadata tests the ExtractInstallerMetadata function. It
+// calls the function for every file under testdata/installers and checks that
+// it returns the expected metadata by comparing it to the software name and
+// version in the filename.
+//
+// The filename should have the following format:
+//
+//	<software_name>$<version>[$<anything>].<extension>
+//
+// That is, it breaks the file name at the dollar sign and the first part is
+// the expected name, the second is the expected version. Note that by default,
+// files in testdata/installers are NOT included in git, so the test files must
+// be added manually (for size and licenses considerations). Why the dollar
+// sign? Because dots, dashes and underlines are more likely to be part of the
+// name or version.
```

Comment:
If you're curious, see test cases and results here: https://github.com/fleetdm/fleet/issues/18318#issuecomment-2075256191